prsa/anchor_point_identification.py

```diff
@@ -0,0 +1,173 @@
+    Identify points in the time series where there is an acceleration, checking against a threshold.
+    The anchor points correspond to increases in the time series that are not too large.
+
+    Parameters:
+    - time_series (np.ndarray): The time series.
+    - threshold (float): The maximum percentage of change between two values acceptable as an acceleration.
+
+    Returns:
+    - np.ndarray: The indices of the detected anchor points.
+    """
+    anchor_points = [idx for idx in range(len(time_series)) if time_series[idx] > time_series[idx - 1]
+                     and abs(time_series[idx] - time_series[idx - 1]) / time_series[idx - 1] < threshold]
+    
+    return np.array(anchor_points)
+
+def detect_anchors_threshold_dc(time_series: np.ndarray, threshold: float) -> np.ndarray:
+    """
+    Identify points in the time series where there is a deceleration, checking against a threshold.
+    The anchor points correspond to decreases in the time series that are not too large.
+
+    Parameters:
+    - time_series (np.ndarray): The time series.
+    - threshold (float): The maximum percentage of change between two values acceptable as a deceleration.
+
+    Returns:
+    - np.ndarray: The indices of the detected anchor points.
+    """
+    anchor_points = [idx for idx in range(len(time_series)) if time_series[idx] < time_series[idx - 1]
+                     and abs(time_series[idx] - time_series[idx - 1]) / time_series[idx - 1] < threshold]
+    
+    return np.array(anchor_points)
+
+def detect_anchors_mean_acc(time_series: np.ndarray, T: int) -> np.ndarray:
+    """
+    Identify acceleration anchor points by comparing averages of T values before and after each point.
+
+    Parameters:
+    - time_series (np.ndarray): The time series.
+    - T (int): The number of values to consider for the average before and after each point.
+
+    Returns:
+    - np.ndarray: The indices of the detected anchor points.
+    """
+    anchor_points = []
+    for idx in range(T, len(time_series) - T):
+        mean_before = np.mean(time_series[idx - T:idx])
+        mean_after = np.mean(time_series[idx:idx + T])
+        if mean_after > mean_before:
+            anchor_points.append(idx)
+    return np.array(anchor_points)
+
+def detect_anchors_mean_dc(time_series: np.ndarray, T: int) -> np.ndarray:
+    """
+    Identify deceleration anchor points by comparing averages of T values before and after each point.
+
+    Parameters:
+    - time_series (np.ndarray): The time series.
+    - T (int): The number of values to consider for the average before and after each point.
+
+    Returns:
+    - np.ndarray: The indices of the detected anchor points.
+    """
+    anchor_points = []
+    for idx in range(T, len(time_series) - T):
+        mean_before = np.mean(time_series[idx - T:idx])
+        mean_after = np.mean(time_series[idx:idx + T])
+        if mean_after < mean_before:
+            anchor_points.append(idx)
+    return np.array(anchor_points)
```

Comment:
komentarz jak wyżej, te też można połączyć, łatwo się pomylić jeśli później używa się funkcji nazwanych `detect_anchors_mean_dc` i `detect_anchors_mean_acc`, dużo lepiej by było z tym wspomnianym argumentem.

Ale jak mówię, to na przyszłość, tutaj niech zostanie jak jest. Do poprawy tylko komentarze do każdej z funkcji, które mylą acc i dc